httpcore/_synchronization.py

```diff
@@ -201,24 +175,14 @@ def __init__(self) -> None:
         self._backend = current_async_library()
 
         if self._backend == "trio":
-            if trio is None:  # pragma: nocover
-                raise RuntimeError(
-                    "Running with trio requires installation of 'httpcore[trio]'."
-                )
-
             self._trio_shield = trio.CancelScope(shield=True)
-        else:
-            if anyio is None:  # pragma: nocover
-                raise RuntimeError(
-                    "Running with asyncio requires installation of 'httpcore[asyncio]'."
-                )
-
+        elif self._backend == "asyncio":
             self._anyio_shield = anyio.CancelScope(shield=True)
 
     def __enter__(self) -> "AsyncShieldCancellation":
         if self._backend == "trio":
             self._trio_shield.__enter__()
-        else:
+        elif self._backend == "asyncio":
             self._anyio_shield.__enter__()
```

Comment:
Should we silently return `self` without doing anything if the backend is somehow invalid? Should there be an `else` that errors instead?

Reply:
Or using `AsyncBackend` enum is proper? we sure only trio and asyncio are acceptable backend.
```py
self._backend: Optional[AsyncBackend] = None
```

Reply:
I think it's *probably* okay enough as-is?
Could we pull this in now and deal with any alternate suggestions as a follow up?

Reply:
Agree, works well and it doesn't disrupt anything btw.